src/main/java/com/example/chat/groupchatbackend/controllers/MessagesController.java

```diff
@@ -24,19 +25,32 @@
     private ConversationsRepository conversationsRepository;
     @Autowired
     private MessagesRepository messagesRepository;
+    @Autowired
+    private ReadStatusRepository readStatusRepository;
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
+    private UserContext userContext;
 
     @GetMapping("/messages/{conversationName}")
     public List<Message> getAllMessages(@PathVariable String conversationName, @RequestParam(required = false) Long timestamp) {
         Conversation conversation = getConversation(conversationName);
 
         if (timestamp == null) {
+            updateReadStatus(conversation, LocalDateTime.now());
             return getMessagesByDate(conversation, LocalDateTime.MIN);
         } else {
             LocalDateTime date = Instant.ofEpochMilli(timestamp).atZone(ZoneId.systemDefault()).toLocalDateTime();
+            updateReadStatus(conversation, date);
             return getMessagesByDate(conversation, date);
         }
     }
 
+    private void updateReadStatus(Conversation conversation, LocalDateTime date) {
```

Comment:
no more comments to the code, one thing to the funtionality - please test that when there is already a ReadStatus row for this Conversation and User , we don't want to add a new one, but update dateTime of the existing one

Reply:
actually we could do this automatically if we removed "id" field for ReadStatus and make `conversationId + userID` combination as our primary key

Reply:
What do you mean by making `conversationId + userID` combination?